app/javascript/packages/form-steps/form-steps.tsx

```diff
@@ -161,14 +167,25 @@ interface FormStepsProps {
  * @param step Current step.
  * @param titleFormat Format string for page title.
  */
-function useStepTitle(step: FormStep, titleFormat?: string) {
+function useStepTitle(step?: FormStep, titleFormat?: string) {
   useEffect(() => {
-    if (titleFormat && step.title) {
+    if (titleFormat && step?.title) {
       document.title = replaceVariables(titleFormat, { step: step.title });
     }
   }, [step]);
 }
 
+function useTrackStepEvents(step?: FormStep) {
+  useEffect(() => {
+    if (step?.name === 'personal_key') {
+      trackEvent('IdV: personal key visited');
+    }
+    if (step?.name === 'personal_key_confirm') {
+      trackEvent('IdV: show personal key modal');
+    }
+  }, [step]);
+}
+
```

Comment:
Can we move this to `VerifyFlow`, via `onStepChange` ? The `FormSteps` implementation should be abstracted such that it wouldn't know anything about these specific step names.